crates/sui-graphql-rpc/src/types/object.rs

```diff
@@ -76,6 +68,34 @@ pub(crate) struct ObjectKey {
     version: u64,
 }
 
+#[derive(Union, Clone)]
+pub enum ObjectOwnerNew {
+    Immutable(Immutable),
+    Shared(Shared),
+    Parent(Parent),
```

Comment:
could just be me but it feels confusing to have Parent here. I'd prefer Child, so on resolution we'd do ... on child { parent { objectField } }

Reply:
I think it's kind of confusing in Rust to call this `Parent`, but it's way way more confusing in the GraphQL schema to have the struct that represent's an object's parent be called `Child`.